### PR TITLE
Generate Help Pipeline Changes

### PR DESCRIPTION
**Pull request by finsharp**
Modify Pipelines to generate docs on DocsGenerationBranch with the format `DocsGeneration\{Date}`.
Create PR with generated docs.

### CODE REVIEW

File: tools/GenerateHelp.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+if (-not (Test-Path $ModuleMappingConfigPath)) {
+    Write-Error "Module mapping file not be found: $ModuleMappingConfigPath."
+}
+if ($ModulesToGenerate.Count -eq 0) {
+    [HashTable] $ModuleMapping = Get-Content $ModuleMappingConfigPath | ConvertFrom-Json -AsHashTable
+    $ModulesToGenerate = $ModuleMapping.Keys
+}
+
+$ModulePrefix = "Microsoft.Graph"
+Import-Module (Join-Path $PSScriptRoot "..\src\Authentication\Authentication\Microsoft.Graph.Authentication.psm1") -Force -Scope Global
```
Comment (peombwa) [inline tools/GenerateHelp.ps1:26]: This is what is causing the build error. It should point to:
```suggestion
Import-Module (Join-Path $PSScriptRoot "..\src\Authentication\Authentication\artifacts\Microsoft.Graph.Authentication.psd1") -Force -Scope Global
```

File: src/Authentication/Authentication.Core/Microsoft.Graph.Authentication.psm1 [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (peombwa) [inline src/Authentication/Authentication.Core/Microsoft.Graph.Authentication.psm1:1]: This file should be removed. Authentication.Core is just a core assembly and not a module.

File: src/Authentication/Authentication/Microsoft.Graph.Authentication.psm1 [hunk 3]
```diff
@@ -1,10 +1,10 @@
 
 # Load the module dll
-$null = Import-Module -Name (Join-Path $PSScriptRoot 'Microsoft.Graph.Authentication.dll')
+$null = Import-Module -Name (Join-Path $PSScriptRoot '.\artifacts\Microsoft.Graph.Authentication.dll')
```
Comment (peombwa) [inline src/Authentication/Authentication/Microsoft.Graph.Authentication.psm1:3]: Changes in this file are not necessary if the first comment is addressed.

File: src/Authentication/Authentication.Core/Microsoft.Graph.Authentication.psm1 (hunk 2 above)
Comment (finsharp) [reply src/Authentication/Authentication.Core/Microsoft.Graph.Authentication.psm1:1]: I assumed the file was from your re-org PR

**Pull request closed without merging**